### PR TITLE
Fix json support for DSS Request Box

### PR DESCRIPTION
**Pull request by madhawa-gunasekara**
Resolves https://github.com/wso2/wso2-axis2/issues/108

### CODE REVIEW

**Comment by madhawa-gunasekara** [comment]
Added integration tests https://github.com/wso2/product-dss/pull/99

File: modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java [hunk 1]
```diff
@@ -268,8 +268,16 @@ public void writeStartElement(String localName) throws XMLStreamException {
...
-                        queObj = queue.peek();
+                        int queueSize = queue.size();
+                        int count = 0;
+                        while (!queObj.getName().equals(localName)) {
+                            if (queue.isEmpty() || count > queueSize) {
+                                break;
+                            }
+                            skipChildElementsFromQueue(queObj);
+                            queObj = queue.peek();
+                            count ++;
```
Comment (jsdjayanga) [inline modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java]: queue.peek() might result null. Hence null check is needed

File: modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java [hunk 2]
```diff
@@ -268,8 +268,16 @@ public void writeStartElement(String localName) throws XMLStreamException {
                     }
                     if (!queObj.getName().equals(localName) && queObj.getMinOccurs() == 0 &&
                         (stack.isEmpty() || !stack.peek().getName().equals(localName))) {
-                        skipChildElementsFromQueue(queObj);
-                        queObj = queue.peek();
+                        int queueSize = queue.size();
+                        int count = 0;
+                        while (!queObj.getName().equals(localName)) {
```
Comment (jsdjayanga) [inline modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java]: queue.peek() might result null. Hence null check is needed

File: modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java [hunk 3]
```diff
@@ -268,8 +268,16 @@ public void writeStartElement(String localName) throws XMLStreamException {
...
+                            if (queue.isEmpty() || count > queueSize) {
```
Comment (jsdjayanga) [inline modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java]: As this queue.isEmpty() is before the other logics, last element in the "queue" won't be called with the skipChildElementsFromQueue(queObj). If the last element having the given local name it won't reach the line 277

Comment (madhawa-gunasekara) [inline modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java]: yes, that's the expected behavior. I have added that logic for stopping criteria.

File: modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java (hunk 2 above)
Comment (madhawa-gunasekara) [inline modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java]: @jsdjayanga this queue is populated from the XmlNodeGenerator class, therefore, there isn't a possibility to have a null object in the queue. but this queue can be empty I have handled it.

**Pull request merged**